demo 报错

### Describe the bug

SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/D:/repo/org/slf4j/slf4j-reload4j/1.7.36/slf4j-reload4j-1.7.36.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/D:/repo/org/apache/logging/log4j/log4j-slf4j-impl/2.17.1/log4j-slf4j-impl-2.17.1.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [org.slf4j.impl.Reload4jLoggerFactory]
log4j:WARN No appenders could be found for logger (org.apache.flink.table.module.ModuleManager).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
Exception in thread "main" org.apache.flink.table.api.ValidationException: Unable to create a sink for writing table 'default_catalog.default_database.t_sink'.

Table options are:

'buffer-flush.batch-size'='100'
'buffer-flush.buffer-size'='5000'
'buffer-flush.interval'='1s'
'compatible-mode'='mysql'
'connection-pool-properties'='druid.initialSize=10;druid.maxActive=100'
'connector'='oceanbase'
'max-retries'='3'
'password'='******'
'table-name'='t_sink'
'upsert-mode'='true'
'url'='jdbc:oceanbase://127.0.0.1:2881/test'
'username'='root@test'
	at org.apache.flink.table.factories.FactoryUtil.createDynamicTableSink(FactoryUtil.java:270)
	at org.apache.flink.table.planner.delegation.PlannerBase.getTableSink(PlannerBase.scala:459)
	at org.apache.flink.table.planner.delegation.PlannerBase.translateToRel(PlannerBase.scala:236)
	at org.apache.flink.table.planner.delegation.PlannerBase.$anonfun$translate$1(PlannerBase.scala:194)
	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:233)
	at scala.collection.Iterator.foreach(Iterator.scala:937)
	at scala.collection.Iterator.foreach$(Iterator.scala:937)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1425)
	at scala.collection.IterableLike.foreach(IterableLike.scala:70)
	at scala.collection.IterableLike.foreach$(IterableLike.scala:69)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
	at scala.collection.TraversableLike.map(TraversableLike.scala:233)
	at scala.collection.TraversableLike.map$(TraversableLike.scala:226)
	at scala.collection.AbstractTraversable.map(Traversable.scala:104)
	at org.apache.flink.table.planner.delegation.PlannerBase.translate(PlannerBase.scala:194)
	at org.apache.flink.table.api.internal.TableEnvironmentImpl.translate(TableEnvironmentImpl.java:1803)
	at org.apache.flink.table.api.internal.TableEnvironmentImpl.executeInternal(TableEnvironmentImpl.java:881)
	at org.apache.flink.table.api.internal.TableEnvironmentImpl.executeInternal(TableEnvironmentImpl.java:989)
	at org.apache.flink.table.api.internal.TableEnvironmentImpl.executeSql(TableEnvironmentImpl.java:765)
	at com.mojie.platform.job.Main.main(Main.java:37)
Caused by: org.apache.flink.table.api.ValidationException: One or more required options are missing.

Missing required options are:

schema-name
	at org.apache.flink.table.factories.FactoryUtil.validateFactoryOptions(FactoryUtil.java:612)
	at org.apache.flink.table.factories.FactoryUtil.validateFactoryOptions(FactoryUtil.java:582)
	at org.apache.flink.table.factories.FactoryUtil$FactoryHelper.validate(FactoryUtil.java:930)
	at com.oceanbase.connector.flink.OceanBaseDynamicTableSinkFactory.createDynamicTableSink(OceanBaseDynamicTableSinkFactory.java:39)
	at org.apache.flink.table.factories.FactoryUtil.createDynamicTableSink(FactoryUtil.java:267)
	... 19 more


### Environment

Flink1.17本地idea

### Fast reproduce steps

把demo复制下来，一运行，就报缺少参数

### Expected behavior

_No response_

### Actual behavior

_No response_

### Additional context

_No response_

--- a/docs/sink/flink-connector-obkv-hbase.md
+++ b/docs/sink/flink-connector-obkv-hbase.md
@@ -113,6 +113,8 @@ public class Main {
 
 Once executed, the records should have been written to OceanBase.
 
+For more information please refer to [OBKVHBaseConnectorITCase.java](../../flink-connector-obkv-hbase/src/test/java/com/oceanbase/connector/flink/OBKVHBaseConnectorITCase.java).
+
 #### Flink SQL Demo
 
 Put the JAR files of dependencies to the 'lib' directory of Flink, and then create the destination table with Flink SQL through the sql client.

--- a/docs/sink/flink-connector-obkv-hbase_cn.md
+++ b/docs/sink/flink-connector-obkv-hbase_cn.md
@@ -113,6 +113,8 @@ public class Main {
 
 执行完成后，即可在 OceanBase 中检索验证。
 
+更多信息请参考 [OBKVHBaseConnectorITCase.java](../../flink-connector-obkv-hbase/src/test/java/com/oceanbase/connector/flink/OBKVHBaseConnectorITCase.java)。
+
 #### Flink SQL 示例
 
 将需要用到的依赖的 JAR 文件放到 Flink 的 lib 目录下，之后通过 SQL Client 在 Flink 中创建目的表。

--- a/docs/sink/flink-connector-oceanbase.md
+++ b/docs/sink/flink-connector-oceanbase.md
@@ -121,6 +121,8 @@ public class Main {
 
 Once executed, the records should have been written to OceanBase.
 
+For more information please refer to [OceanBaseConnectorITCase.java](../../flink-connector-oceanbase/src/test/java/com/oceanbase/connector/flink/OceanBaseConnectorITCase.java).
+
 #### Flink SQL Demo
 
 Put the JAR files of dependencies to the 'lib' directory of Flink, and then create the destination table with Flink SQL through the sql client.

--- a/docs/sink/flink-connector-oceanbase.md
+++ b/docs/sink/flink-connector-oceanbase.md
@@ -95,8 +95,9 @@ public class Main {
                         + ") with ("
                         + "    'connector' = 'oceanbase',"
                         + "    'url' = 'jdbc:oceanbase://127.0.0.1:2881/test',"
+                        + "    'schema-name'= 'test',"
                         + "    'table-name' = 't_sink',"
-                        + "    'username' = 'root@test',"
+                        + "    'username' = 'root@test#obcluster',"
                         + "    'password' = 'pswd',"
                         + "    'compatible-mode' = 'mysql',"
                         + "    'connection-pool-properties' = 'druid.initialSize=10;druid.maxActive=100',"
@@ -115,6 +116,7 @@ public class Main {
                 .await();
     }
 }
+
 ```
 
 Once executed, the records should have been written to OceanBase.
@@ -133,8 +135,6 @@ CREATE TABLE t_sink
 ) with (
       'connector' = 'oceanbase',
       'url' = 'jdbc:oceanbase://127.0.0.1:2881/test',
-      'cluster-name' = 'obcluster',
-      'tenant-name' = 'test',
       'schema-name' = 'test',
       'table-name' = 't_sink',
       'username' = 'root@test#obcluster',

--- a/docs/sink/flink-connector-oceanbase_cn.md
+++ b/docs/sink/flink-connector-oceanbase_cn.md
@@ -122,6 +122,8 @@ public class Main {
 
 执行完成后，即可在 OceanBase 中检索验证。
 
+更多信息请参考 [OceanBaseConnectorITCase.java](../../flink-connector-oceanbase/src/test/java/com/oceanbase/connector/flink/OceanBaseConnectorITCase.java)。
+
 #### Flink SQL 示例
 
 将需要用到的依赖的 JAR 文件放到 Flink 的 lib 目录下，之后通过 SQL Client 在 Flink 中创建目的表。

--- a/docs/sink/flink-connector-oceanbase_cn.md
+++ b/docs/sink/flink-connector-oceanbase_cn.md
@@ -96,8 +96,9 @@ public class Main {
                         + ") with ("
                         + "    'connector' = 'oceanbase',"
                         + "    'url' = 'jdbc:oceanbase://127.0.0.1:2881/test',"
+                        + "    'schema-name'= 'test',"
                         + "    'table-name' = 't_sink',"
-                        + "    'username' = 'root@test',"
+                        + "    'username' = 'root@test#obcluster',"
                         + "    'password' = 'pswd',"
                         + "    'compatible-mode' = 'mysql',"
                         + "    'connection-pool-properties' = 'druid.initialSize=10;druid.maxActive=100',"
@@ -116,6 +117,7 @@ public class Main {
                 .await();
     }
 }
+
 ```
 
 执行完成后，即可在 OceanBase 中检索验证。
@@ -135,8 +137,6 @@ CREATE TABLE t_sink
 with (
     'connector' = 'oceanbase',
     'url' = 'jdbc:oceanbase://127.0.0.1:2881/test',
-    'cluster-name' = 'obcluster',
-    'tenant-name' = 'test',
     'schema-name' = 'test',
     'table-name' = 't_sink',
     'username' = 'root@test#obcluster',

--- a/flink-connector-obkv-hbase/src/test/java/com/oceanbase/connector/flink/OBKVHBaseConnectorITCase.java
+++ b/flink-connector-obkv-hbase/src/test/java/com/oceanbase/connector/flink/OBKVHBaseConnectorITCase.java
@@ -165,7 +165,10 @@ public class OBKVHBaseConnectorITCase extends OceanBaseTestBase {
     }
 
     private RowData rowData(String rowKey, Integer q1, String q2, Integer q3) {
-        return GenericRowData.of(StringData.fromString(rowKey), q1, StringData.fromString(q2), q3);
+        return GenericRowData.of(
+                StringData.fromString(rowKey),
+                GenericRowData.of(q1),
+                GenericRowData.of(StringData.fromString(q2), q3));
     }
 
     @Test

--- a/flink-connector-obkv-hbase/src/test/java/com/oceanbase/connector/flink/OBKVHBaseConnectorITCase.java
+++ b/flink-connector-obkv-hbase/src/test/java/com/oceanbase/connector/flink/OBKVHBaseConnectorITCase.java
@@ -16,9 +16,21 @@
 
 package com.oceanbase.connector.flink;
 
+import com.oceanbase.connector.flink.connection.OBKVHBaseConnectionProvider;
+import com.oceanbase.connector.flink.connection.OBKVHBaseTableSchema;
+import com.oceanbase.connector.flink.sink.OBKVHBaseStatementExecutor;
+import com.oceanbase.connector.flink.sink.OceanBaseSink;
+
 import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.table.api.DataTypes;
 import org.apache.flink.table.api.EnvironmentSettings;
 import org.apache.flink.table.api.bridge.java.StreamTableEnvironment;
+import org.apache.flink.table.catalog.Column;
+import org.apache.flink.table.catalog.ResolvedSchema;
+import org.apache.flink.table.catalog.UniqueConstraint;
+import org.apache.flink.table.data.GenericRowData;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.data.StringData;
 
 import com.alipay.oceanbase.hbase.OHTableClient;
 import com.alipay.oceanbase.hbase.constants.OHConstants;
@@ -37,6 +49,7 @@ import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collections;
 import java.util.List;
+import java.util.Map;
 
 import static org.junit.Assert.assertTrue;
 
@@ -45,6 +58,11 @@ public class OBKVHBaseConnectorITCase extends OceanBaseTestBase {
     public static final String CLUSTER_NAME = "obcluster";
     public static final String CONFIG_URL =
             "http://127.0.0.1:8080/services?Action=ObRootServiceInfo&ObCluster=" + CLUSTER_NAME;
+
+    @Override
+    protected String getTestTable() {
+        return "htable";
+    }
 
     @Override
     protected String getUrl() {
@@ -57,8 +75,11 @@ public class OBKVHBaseConnectorITCase extends OceanBaseTestBase {
     }
 
     @Override
-    protected String getTestTable() {
-        return "htable";
+    protected Map<String, String> getOptions() {
+        Map<String, String> options = super.getOptions();
+        options.put("sys.username", OB_SERVER.getSysUsername());
+        options.put("sys.password", OB_SERVER.getSysPassword());
+        return options;
     }
 
     private OHTableClient client;
@@ -94,6 +115,60 @@ public class OBKVHBaseConnectorITCase extends OceanBaseTestBase {
     }
 
     @Test
+    public void testDataStreamSink() throws Exception {
+        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+        env.setParallelism(1);
+
+        ResolvedSchema physicalSchema =
+                new ResolvedSchema(
+                        Arrays.asList(
+                                Column.physical("rowkey", DataTypes.STRING().notNull()),
+                                Column.physical(
+                                        "family1",
+                                        DataTypes.ROW(
+                                                        DataTypes.FIELD(
+                                                                "q1", DataTypes.INT().nullable()))
+                                                .notNull()),
+                                Column.physical(
+                                        "family2",
+                                        DataTypes.ROW(
+                                                        DataTypes.FIELD(
+                                                                "q2",
+                                                                DataTypes.STRING().nullable()),
+                                                        DataTypes.FIELD(
+                                                                "q3", DataTypes.INT().nullable()))
+                                                .notNull())),
+                        Collections.emptyList(),
+                        UniqueConstraint.primaryKey("pk", Collections.singletonList("rowkey")));
+
+        List<RowData> dataSet =
+                Arrays.asList(
+                        rowData("1", 1, "1", 1),
+                        rowData("2", null, "2", null),
+                        rowData("3", 3, null, null),
+                        rowData("4", 4, "4", null));
+
+        OBKVHBaseConnectorOptions connectorOptions = new OBKVHBaseConnectorOptions(getOptions());
+        OBKVHBaseConnectionProvider connectionProvider =
+                new OBKVHBaseConnectionProvider(connectorOptions.getConnectionOptions());
+        OBKVHBaseStatementExecutor statementExecutor =
+                new OBKVHBaseStatementExecutor(
+                        connectorOptions.getStatementOptions(),
+                        new OBKVHBaseTableSchema(physicalSchema),
+                        connectionProvider);
+        OceanBaseSink sink =
+                new OceanBaseSink(connectorOptions.getWriterOptions(), statementExecutor);
+        env.fromCollection(dataSet).sinkTo(sink);
+        env.execute();
+
+        validateSinkResults();
+    }
+
+    private RowData rowData(String rowKey, Integer q1, String q2, Integer q3) {
+        return GenericRowData.of(StringData.fromString(rowKey), q1, StringData.fromString(q2), q3);
+    }
+
+    @Test
     public void testSink() throws Exception {
         StreamExecutionEnvironment execEnv = StreamExecutionEnvironment.getExecutionEnvironment();
         execEnv.setParallelism(1);
@@ -102,20 +177,15 @@ public class OBKVHBaseConnectorITCase extends OceanBaseTestBase {
                         execEnv, EnvironmentSettings.newInstance().inStreamingMode().build());
 
         tEnv.executeSql(
-                String.format(
-                        "CREATE TEMPORARY TABLE target ("
-                                + " rowkey STRING,"
-                                + " family1 ROW<q1 INT>,"
-                                + " family2 ROW<q2 STRING, q3 INT>,"
-                                + " PRIMARY KEY (rowkey) NOT ENFORCED"
-                                + ") with ("
-                                + "  'connector'='obkv-hbase',"
-                                + "  'sys.username'='%s',"
-                                + "  'sys.password'='%s',"
-                                + getCommonOptionsString()
-                                + ");",
-                        OB_SERVER.getSysUsername(),
-                        OB_SERVER.getSysPassword()));
+                "CREATE TEMPORARY TABLE target ("
+                        + " rowkey STRING,"
+                        + " family1 ROW<q1 INT>,"
+                        + " family2 ROW<q2 STRING, q3 INT>,"
+                        + " PRIMARY KEY (rowkey) NOT ENFORCED"
+                        + ") with ("
+                        + "  'connector'='obkv-hbase',"
+                        + getOptionsString()
+                        + ");");
 
         tEnv.executeSql(
                         String.format(

--- a/flink-connector-obkv-hbase/src/test/java/com/oceanbase/connector/flink/OBKVHBaseConnectorITCase.java
+++ b/flink-connector-obkv-hbase/src/test/java/com/oceanbase/connector/flink/OBKVHBaseConnectorITCase.java
@@ -44,6 +44,21 @@ public class OBKVHBaseConnectorITCase extends OceanBaseTestBase {
     public static final String CONFIG_URL =
             "http://127.0.0.1:8080/services?Action=ObRootServiceInfo&ObCluster=" + CLUSTER_NAME;
 
+    @Override
+    protected String getUrl() {
+        return String.format("%s&database=%s", CONFIG_URL, OB_SERVER.getDatabaseName());
+    }
+
+    @Override
+    protected String getUsername() {
+        return OB_SERVER.getUsername() + "#" + CLUSTER_NAME;
+    }
+
+    @Override
+    protected String getTestTable() {
+        return "htable";
+    }
+
     @Test
     public void testSink() throws Exception {
         StreamExecutionEnvironment execEnv = StreamExecutionEnvironment.getExecutionEnvironment();
@@ -52,12 +67,8 @@ public class OBKVHBaseConnectorITCase extends OceanBaseTestBase {
                 StreamTableEnvironment.create(
                         execEnv, EnvironmentSettings.newInstance().inStreamingMode().build());
 
-        String hTable = "htable";
         String family1 = "family1";
         String family2 = "family2";
-
-        String url = String.format("%s&database=%s", CONFIG_URL, obServer.getDatabaseName());
-        String fullUsername = obServer.getUsername() + "#" + CLUSTER_NAME;
 
         tEnv.executeSql(
                 String.format(
@@ -68,21 +79,14 @@ public class OBKVHBaseConnectorITCase extends OceanBaseTestBase {
                                 + " PRIMARY KEY (rowkey) NOT ENFORCED"
                                 + ") with ("
                                 + "  'connector'='obkv-hbase',"
-                                + "  'url'='%s',"
-                                + "  'table-name'='%s',"
-                                + "  'username'='%s',"
-                                + "  'password'='%s',"
                                 + "  'sys.username'='%s',"
-                                + "  'sys.password'='%s'"
+                                + "  'sys.password'='%s',"
+                                + getCommonOptionsString()
                                 + ");",
                         family1,
                         family2,
-                        url,
-                        hTable,
-                        fullUsername,
-                        obServer.getPassword(),
-                        obServer.getSysUsername(),
-                        obServer.getSysPassword()));
+                        OB_SERVER.getSysUsername(),
+                        OB_SERVER.getSysPassword()));
 
         tEnv.executeSql(
                         String.format(
@@ -94,13 +98,13 @@ public class OBKVHBaseConnectorITCase extends OceanBaseTestBase {
                 .await();
 
         Configuration conf = new Configuration();
-        conf.set(OHConstants.HBASE_OCEANBASE_PARAM_URL, url);
-        conf.set(OHConstants.HBASE_OCEANBASE_FULL_USER_NAME, fullUsername);
-        conf.set(OHConstants.HBASE_OCEANBASE_PASSWORD, obServer.getPassword());
-        conf.set(OHConstants.HBASE_OCEANBASE_SYS_USER_NAME, obServer.getSysUsername());
-        conf.set(OHConstants.HBASE_OCEANBASE_SYS_PASSWORD, obServer.getSysPassword());
+        conf.set(OHConstants.HBASE_OCEANBASE_PARAM_URL, getUrl());
+        conf.set(OHConstants.HBASE_OCEANBASE_FULL_USER_NAME, getUsername());
+        conf.set(OHConstants.HBASE_OCEANBASE_PASSWORD, getPassword());
+        conf.set(OHConstants.HBASE_OCEANBASE_SYS_USER_NAME, OB_SERVER.getSysUsername());
+        conf.set(OHConstants.HBASE_OCEANBASE_SYS_PASSWORD, OB_SERVER.getSysPassword());
 
-        OHTableClient client = new OHTableClient(hTable, conf);
+        OHTableClient client = new OHTableClient(getTestTable(), conf);
         client.init();
 
         Function<KeyValue, String> valueFunc =

--- a/flink-connector-obkv-hbase/src/test/java/com/oceanbase/connector/flink/OBKVHBaseConnectorITCase.java
+++ b/flink-connector-obkv-hbase/src/test/java/com/oceanbase/connector/flink/OBKVHBaseConnectorITCase.java
@@ -24,9 +24,12 @@ import com.alipay.oceanbase.hbase.OHTableClient;
 import com.alipay.oceanbase.hbase.constants.OHConstants;
 import org.apache.hadoop.conf.Configuration;
 import org.apache.hadoop.hbase.KeyValue;
+import org.apache.hadoop.hbase.client.Delete;
 import org.apache.hadoop.hbase.client.Get;
 import org.apache.hadoop.hbase.client.Result;
 import org.apache.hadoop.hbase.util.Bytes;
+import org.junit.After;
+import org.junit.Before;
 import org.junit.Test;
 
 import java.io.IOException;
@@ -59,6 +62,32 @@ public class OBKVHBaseConnectorITCase extends OceanBaseTestBase {
         return "htable";
     }
 
+    private OHTableClient client;
+
+    @Before
+    public void before() throws Exception {
+        Configuration conf = new Configuration();
+        conf.set(OHConstants.HBASE_OCEANBASE_PARAM_URL, getUrl());
+        conf.set(OHConstants.HBASE_OCEANBASE_FULL_USER_NAME, getUsername());
+        conf.set(OHConstants.HBASE_OCEANBASE_PASSWORD, getPassword());
+        conf.set(OHConstants.HBASE_OCEANBASE_SYS_USER_NAME, OB_SERVER.getSysUsername());
+        conf.set(OHConstants.HBASE_OCEANBASE_SYS_PASSWORD, OB_SERVER.getSysPassword());
+        client = new OHTableClient(getTestTable(), conf);
+        client.init();
+    }
+
+    @After
+    public void after() throws Exception {
+        client.delete(
+                Arrays.asList(
+                        new Delete(Bytes.toBytes("1")),
+                        new Delete(Bytes.toBytes("2")),
+                        new Delete(Bytes.toBytes("3")),
+                        new Delete(Bytes.toBytes("4"))));
+        client.close();
+        client = null;
+    }
+
     @Test
     public void testSink() throws Exception {
         StreamExecutionEnvironment execEnv = StreamExecutionEnvironment.getExecutionEnvironment();
@@ -67,15 +96,12 @@ public class OBKVHBaseConnectorITCase extends OceanBaseTestBase {
                 StreamTableEnvironment.create(
                         execEnv, EnvironmentSettings.newInstance().inStreamingMode().build());
 
-        String family1 = "family1";
-        String family2 = "family2";
-
         tEnv.executeSql(
                 String.format(
                         "CREATE TEMPORARY TABLE target ("
                                 + " rowkey STRING,"
-                                + " %s ROW<q1 INT>,"
-                                + " %s ROW<q2 STRING, q3 INT>,"
+                                + " family1 ROW<q1 INT>,"
+                                + " family2 ROW<q2 STRING, q3 INT>,"
                                 + " PRIMARY KEY (rowkey) NOT ENFORCED"
                                 + ") with ("
                                 + "  'connector'='obkv-hbase',"
@@ -83,8 +109,6 @@ public class OBKVHBaseConnectorITCase extends OceanBaseTestBase {
                                 + "  'sys.password'='%s',"
                                 + getCommonOptionsString()
                                 + ");",
-                        family1,
-                        family2,
                         OB_SERVER.getSysUsername(),
                         OB_SERVER.getSysPassword()));
 
@@ -97,16 +121,10 @@ public class OBKVHBaseConnectorITCase extends OceanBaseTestBase {
                                 row("4", 4, "4", null)))
                 .await();
 
-        Configuration conf = new Configuration();
-        conf.set(OHConstants.HBASE_OCEANBASE_PARAM_URL, getUrl());
-        conf.set(OHConstants.HBASE_OCEANBASE_FULL_USER_NAME, getUsername());
-        conf.set(OHConstants.HBASE_OCEANBASE_PASSWORD, getPassword());
-        conf.set(OHConstants.HBASE_OCEANBASE_SYS_USER_NAME, OB_SERVER.getSysUsername());
-        conf.set(OHConstants.HBASE_OCEANBASE_SYS_PASSWORD, OB_SERVER.getSysPassword());
+        validateSinkResults();
+    }
 
-        OHTableClient client = new OHTableClient(getTestTable(), conf);
-        client.init();
-
+    private void validateSinkResults() throws Exception {
         Function<KeyValue, String> valueFunc =
                 kv -> {
                     String column = Bytes.toString(kv.getQualifier());
@@ -118,29 +136,23 @@ public class OBKVHBaseConnectorITCase extends OceanBaseTestBase {
                 };
 
         assertEqualsInAnyOrder(
-                Collections.singletonList("1,q1,1"), queryHTable(client, family1, "1", valueFunc));
-        assertTrue(queryHTable(client, family1, "2", valueFunc).isEmpty());
+                Collections.singletonList("1,q1,1"), queryHTable(client, "family1", "1"));
+        assertTrue(queryHTable(client, "family1", "2").isEmpty());
         assertEqualsInAnyOrder(
-                Collections.singletonList("3,q1,3"), queryHTable(client, family1, "3", valueFunc));
+                Collections.singletonList("3,q1,3"), queryHTable(client, "family1", "3"));
         assertEqualsInAnyOrder(
-                Collections.singletonList("4,q1,4"), queryHTable(client, family1, "4", valueFunc));
+                Collections.singletonList("4,q1,4"), queryHTable(client, "family1", "4"));
 
         assertEqualsInAnyOrder(
-                Arrays.asList("1,q2,1", "1,q3,1"), queryHTable(client, family2, "1", valueFunc));
+                Arrays.asList("1,q2,1", "1,q3,1"), queryHTable(client, "family2", "1"));
         assertEqualsInAnyOrder(
-                Collections.singletonList("2,q2,2"), queryHTable(client, family2, "2", valueFunc));
-        assertTrue(queryHTable(client, family2, "3", valueFunc).isEmpty());
+                Collections.singletonList("2,q2,2"), queryHTable(client, "family2", "2"));
+        assertTrue(queryHTable(client, "family2", "3").isEmpty());
         assertEqualsInAnyOrder(
-                Collections.singletonList("4,q2,4"), queryHTable(client, family2, "4", valueFunc));
-
-        client.close();
+                Collections.singletonList("4,q2,4"), queryHTable(client, "family2", "4"));
     }
 
-    private List<String> queryHTable(
-            OHTableClient client,
-            String family,
-            String rowKey,
-            Function<KeyValue, String> valueStringFunction)
+    private List<String> queryHTable(OHTableClient client, String family, String rowKey)
             throws IOException {
         List<String> result = new ArrayList<>();
         Get get = new Get(Bytes.toBytes(rowKey));
@@ -150,12 +162,15 @@ public class OBKVHBaseConnectorITCase extends OceanBaseTestBase {
             return result;
         }
         for (KeyValue kv : r.list()) {
+            String column = Bytes.toString(kv.getQualifier());
             result.add(
                     String.format(
                             "%s,%s,%s",
                             rowKey,
-                            Bytes.toString(kv.getQualifier()),
-                            valueStringFunction.apply(kv)));
+                            column,
+                            "q2".equals(column)
+                                    ? Bytes.toString(kv.getValue())
+                                    : String.valueOf(Bytes.toInt(kv.getValue()))));
         }
         return result;
     }

--- a/flink-connector-obkv-hbase/src/test/java/com/oceanbase/connector/flink/OBKVHBaseConnectorITCase.java
+++ b/flink-connector-obkv-hbase/src/test/java/com/oceanbase/connector/flink/OBKVHBaseConnectorITCase.java
@@ -37,7 +37,6 @@ import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collections;
 import java.util.List;
-import java.util.function.Function;
 
 import static org.junit.Assert.assertTrue;
 
@@ -80,12 +79,18 @@ public class OBKVHBaseConnectorITCase extends OceanBaseTestBase {
     public void after() throws Exception {
         client.delete(
                 Arrays.asList(
-                        new Delete(Bytes.toBytes("1")),
-                        new Delete(Bytes.toBytes("2")),
-                        new Delete(Bytes.toBytes("3")),
-                        new Delete(Bytes.toBytes("4"))));
+                        deleteFamily("1", "family1"),
+                        deleteFamily("1", "family2"),
+                        deleteFamily("2", "family2"),
+                        deleteFamily("3", "family1"),
+                        deleteFamily("4", "family1"),
+                        deleteFamily("4", "family2")));
         client.close();
         client = null;
+    }
+
+    private Delete deleteFamily(String rowKey, String family) {
+        return new Delete(Bytes.toBytes(rowKey)).deleteFamily(Bytes.toBytes(family));
     }
 
     @Test
@@ -125,16 +130,6 @@ public class OBKVHBaseConnectorITCase extends OceanBaseTestBase {
     }
 
     private void validateSinkResults() throws Exception {
-        Function<KeyValue, String> valueFunc =
-                kv -> {
-                    String column = Bytes.toString(kv.getQualifier());
-                    if ("q2".equals(column)) {
-                        return Bytes.toString(kv.getValue());
-                    } else {
-                        return String.valueOf(Bytes.toInt(kv.getValue()));
-                    }
-                };
-
         assertEqualsInAnyOrder(
                 Collections.singletonList("1,q1,1"), queryHTable(client, "family1", "1"));
         assertTrue(queryHTable(client, "family1", "2").isEmpty());

--- a/flink-connector-obkv-hbase/src/test/resources/sql/init.sql
+++ b/flink-connector-obkv-hbase/src/test/resources/sql/init.sql
@@ -11,7 +11,8 @@
 -- specific language governing permissions and limitations
 -- under the License.
 
-use test;
+CREATE DATABASE IF NOT EXISTS test;
+USE test;
 
 CREATE TABLE `htable$family1`
 (

--- a/flink-connector-oceanbase-base/src/test/java/com/oceanbase/connector/flink/OceanBaseTestBase.java
+++ b/flink-connector-oceanbase-base/src/test/java/com/oceanbase/connector/flink/OceanBaseTestBase.java
@@ -63,7 +63,7 @@ public abstract class OceanBaseTestBase extends TestLogger {
         return OB_SERVER.getPassword();
     }
 
-    protected Map<String, String> getCommonOptions() {
+    protected Map<String, String> getOptions() {
         Map<String, String> options = new HashMap<>();
         options.put("url", getUrl());
         options.put("table-name", getTestTable());
@@ -72,8 +72,8 @@ public abstract class OceanBaseTestBase extends TestLogger {
         return options;
     }
 
-    protected String getCommonOptionsString() {
-        return getCommonOptions().entrySet().stream()
+    protected String getOptionsString() {
+        return getOptions().entrySet().stream()
                 .map(e -> String.format("'%s'='%s'", e.getKey(), e.getValue()))
                 .collect(Collectors.joining(","));
     }

--- a/flink-connector-oceanbase-base/src/test/java/com/oceanbase/connector/flink/OceanBaseTestBase.java
+++ b/flink-connector-oceanbase-base/src/test/java/com/oceanbase/connector/flink/OceanBaseTestBase.java
@@ -18,54 +18,71 @@ package com.oceanbase.connector.flink;
 
 import org.apache.flink.util.TestLogger;
 
-import org.junit.After;
-import org.junit.Before;
+import org.junit.ClassRule;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.testcontainers.containers.output.Slf4jLogConsumer;
-import org.testcontainers.lifecycle.Startables;
 import org.testcontainers.utility.MountableFile;
 
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 import java.util.stream.Collectors;
 
 import static org.junit.Assert.assertArrayEquals;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertTrue;
 
-public class OceanBaseTestBase extends TestLogger {
+public abstract class OceanBaseTestBase extends TestLogger {
 
     private static final Logger LOG = LoggerFactory.getLogger(OceanBaseTestBase.class);
 
-    protected OceanBaseContainer obServer;
-    protected String imageTag = "4.2.1_bp2";
+    public static final String IMAGE_TAG = "4.2.1_bp2";
 
-    @Before
-    public void before() {
-        obServer =
-                new OceanBaseContainer(OceanBaseContainer.DOCKER_IMAGE_NAME + ":" + imageTag)
-                        .withNetworkMode("host")
-                        .withSysPassword("123456")
-                        .withCopyFileToContainer(
-                                MountableFile.forClasspathResource("sql/init.sql"),
-                                "/root/boot/init.d/init.sql")
-                        .withLogConsumer(new Slf4jLogConsumer(LOG));
-
-        Startables.deepStart(obServer).join();
-    }
-
-    @After
-    public void after() {
-        if (obServer != null) {
-            obServer.close();
-        }
-    }
+    @ClassRule
+    public static final OceanBaseContainer OB_SERVER =
+            new OceanBaseContainer(OceanBaseContainer.DOCKER_IMAGE_NAME + ":" + IMAGE_TAG)
+                    .withNetworkMode("host")
+                    .withSysPassword("123456")
+                    .withCopyFileToContainer(
+                            MountableFile.forClasspathResource("sql/init.sql"),
+                            "/root/boot/init.d/init.sql")
+                    .withLogConsumer(new Slf4jLogConsumer(LOG));
 
     public static void assertEqualsInAnyOrder(List<String> expected, List<String> actual) {
         assertTrue(expected != null && actual != null);
         assertEqualsInOrder(
                 expected.stream().sorted().collect(Collectors.toList()),
                 actual.stream().sorted().collect(Collectors.toList()));
+    }
+
+    protected String getUrl() {
+        return OB_SERVER.getJdbcUrl();
+    }
+
+    protected abstract String getTestTable();
+
+    protected String getUsername() {
+        return OB_SERVER.getUsername();
+    }
+
+    protected String getPassword() {
+        return OB_SERVER.getPassword();
+    }
+
+    protected Map<String, String> getCommonOptions() {
+        Map<String, String> options = new HashMap<>();
+        options.put("url", getUrl());
+        options.put("table-name", getTestTable());
+        options.put("username", getUsername());
+        options.put("password", getPassword());
+        return options;
+    }
+
+    protected String getCommonOptionsString() {
+        return getCommonOptions().entrySet().stream()
+                .map(e -> String.format("'%s'='%s'", e.getKey(), e.getValue()))
+                .collect(Collectors.joining(","));
     }
 
     public static void assertEqualsInOrder(List<String> expected, List<String> actual) {

--- a/flink-connector-oceanbase-base/src/test/java/com/oceanbase/connector/flink/OceanBaseTestBase.java
+++ b/flink-connector-oceanbase-base/src/test/java/com/oceanbase/connector/flink/OceanBaseTestBase.java
@@ -49,13 +49,6 @@ public abstract class OceanBaseTestBase extends TestLogger {
                             "/root/boot/init.d/init.sql")
                     .withLogConsumer(new Slf4jLogConsumer(LOG));
 
-    public static void assertEqualsInAnyOrder(List<String> expected, List<String> actual) {
-        assertTrue(expected != null && actual != null);
-        assertEqualsInOrder(
-                expected.stream().sorted().collect(Collectors.toList()),
-                actual.stream().sorted().collect(Collectors.toList()));
-    }
-
     protected String getUrl() {
         return OB_SERVER.getJdbcUrl();
     }
@@ -83,6 +76,13 @@ public abstract class OceanBaseTestBase extends TestLogger {
         return getCommonOptions().entrySet().stream()
                 .map(e -> String.format("'%s'='%s'", e.getKey(), e.getValue()))
                 .collect(Collectors.joining(","));
+    }
+
+    public static void assertEqualsInAnyOrder(List<String> expected, List<String> actual) {
+        assertTrue(expected != null && actual != null);
+        assertEqualsInOrder(
+                expected.stream().sorted().collect(Collectors.toList()),
+                actual.stream().sorted().collect(Collectors.toList()));
     }
 
     public static void assertEqualsInOrder(List<String> expected, List<String> actual) {

--- a/flink-connector-oceanbase/src/test/java/com/oceanbase/connector/flink/OceanBaseConnectorITCase.java
+++ b/flink-connector-oceanbase/src/test/java/com/oceanbase/connector/flink/OceanBaseConnectorITCase.java
@@ -16,12 +16,28 @@
 
 package com.oceanbase.connector.flink;
 
+import com.oceanbase.connector.flink.connection.OceanBaseConnectionPool;
+import com.oceanbase.connector.flink.connection.OceanBaseConnectionProvider;
+import com.oceanbase.connector.flink.connection.OceanBaseTableSchema;
+import com.oceanbase.connector.flink.sink.OceanBaseSink;
+import com.oceanbase.connector.flink.sink.OceanBaseStatementExecutor;
+
 import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.table.api.DataTypes;
 import org.apache.flink.table.api.EnvironmentSettings;
 import org.apache.flink.table.api.bridge.java.StreamTableEnvironment;
+import org.apache.flink.table.catalog.Column;
+import org.apache.flink.table.catalog.ResolvedSchema;
+import org.apache.flink.table.catalog.UniqueConstraint;
+import org.apache.flink.table.data.DecimalData;
+import org.apache.flink.table.data.GenericRowData;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.data.StringData;
 
+import org.junit.After;
 import org.junit.Test;
 
+import java.math.BigDecimal;
 import java.sql.Connection;
 import java.sql.DriverManager;
 import java.sql.ResultSet;
@@ -30,43 +46,109 @@ import java.sql.SQLException;
 import java.sql.Statement;
 import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.Collections;
 import java.util.List;
+import java.util.Map;
 
 public class OceanBaseConnectorITCase extends OceanBaseTestBase {
 
+    @Override
+    protected String getTestTable() {
+        return "products";
+    }
+
+    @Override
+    protected Map<String, String> getCommonOptions() {
+        Map<String, String> options = super.getCommonOptions();
+        options.put("compatible-mode", "mysql");
+        options.put("schema-name", "test");
+        options.put("connection-pool-properties", "druid.initialSize=4;druid.maxActive=20;");
+        return options;
+    }
+
+    @After
+    public void after() throws Exception {
+        try (Connection connection = getConnection();
+                Statement statement = connection.createStatement()) {
+            statement.execute("DELETE FROM " + getTestTable());
+        }
+    }
+
+    @Test
+    public void testDataStreamSink() throws Exception {
+        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+        env.setParallelism(1);
+
+        ResolvedSchema physicalSchema =
+                new ResolvedSchema(
+                        Arrays.asList(
+                                Column.physical("id", DataTypes.INT().notNull()),
+                                Column.physical("name", DataTypes.STRING().notNull()),
+                                Column.physical("description", DataTypes.STRING().notNull()),
+                                Column.physical("weight", DataTypes.DECIMAL(20, 10).notNull())),
+                        Collections.emptyList(),
+                        UniqueConstraint.primaryKey("pk", Collections.singletonList("id")));
+
+        List<RowData> dataSet =
+                Arrays.asList(
+                        rowData(101, "scooter", "Small 2-wheel scooter", 3.14),
+                        rowData(102, "car battery", "12V car battery", 8.1),
+                        rowData(
+                                103,
+                                "12-pack drill bits",
+                                "12-pack of drill bits with sizes ranging from #40 to #3",
+                                0.8),
+                        rowData(104, "hammer", "12oz carpenter's hammer", 0.75),
+                        rowData(105, "hammer", "14oz carpenter's hammer", 0.875),
+                        rowData(106, "hammer", "16oz carpenter's hammer", 1.0),
+                        rowData(107, "rocks", "box of assorted rocks", 5.3),
+                        rowData(108, "jacket", "water resistent black wind breaker", 0.1),
+                        rowData(109, "spare tire", "24 inch spare tire", 22.2));
+
+        OceanBaseConnectorOptions connectorOptions =
+                new OceanBaseConnectorOptions(getCommonOptions());
+
+        OceanBaseConnectionProvider connectionProvider =
+                new OceanBaseConnectionPool(connectorOptions.getConnectionOptions());
+        OceanBaseTableSchema tableSchema = new OceanBaseTableSchema(physicalSchema);
+        OceanBaseStatementExecutor statementExecutor =
+                new OceanBaseStatementExecutor(
+                        connectorOptions.getStatementOptions(), tableSchema, connectionProvider);
+        OceanBaseSink sink =
+                new OceanBaseSink(connectorOptions.getWriterOptions(), statementExecutor);
+        env.fromCollection(dataSet).sinkTo(sink);
+        env.execute();
+
+        validateSinkResults();
+    }
+
+    private RowData rowData(int id, String name, String description, double weight) {
+        return GenericRowData.of(
+                id,
+                StringData.fromString(name),
+                StringData.fromString(description),
+                DecimalData.fromBigDecimal(new BigDecimal(weight), 20, 10));
+    }
+
     @Test
     public void testSink() throws Exception {
-        StreamExecutionEnvironment execEnv = StreamExecutionEnvironment.getExecutionEnvironment();
-        execEnv.setParallelism(1);
+        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+        env.setParallelism(1);
         StreamTableEnvironment tEnv =
                 StreamTableEnvironment.create(
-                        execEnv, EnvironmentSettings.newInstance().inStreamingMode().build());
-
-        String tableName = "products";
+                        env, EnvironmentSettings.newInstance().inStreamingMode().build());
 
         tEnv.executeSql(
-                String.format(
-                        "CREATE TEMPORARY TABLE target ("
-                                + " `id` INT NOT NULL,"
-                                + " name STRING,"
-                                + " description STRING,"
-                                + " weight DECIMAL(20, 10),"
-                                + " PRIMARY KEY (`id`) NOT ENFORCED"
-                                + ") with ("
-                                + "  'connector'='oceanbase',"
-                                + "  'url'='%s',"
-                                + "  'schema-name'='%s',"
-                                + "  'table-name'='%s',"
-                                + "  'username'='%s',"
-                                + "  'password'='%s',"
-                                + "  'compatible-mode'='mysql',"
-                                + "  'connection-pool-properties'='druid.initialSize=4;druid.maxActive=20;'"
-                                + ");",
-                        obServer.getJdbcUrl(),
-                        obServer.getDatabaseName(),
-                        tableName,
-                        obServer.getUsername(),
-                        obServer.getPassword()));
+                "CREATE TEMPORARY TABLE target ("
+                        + " `id` INT NOT NULL,"
+                        + " name STRING,"
+                        + " description STRING,"
+                        + " weight DECIMAL(20, 10),"
+                        + " PRIMARY KEY (`id`) NOT ENFORCED"
+                        + ") with ("
+                        + "  'connector'='oceanbase',"
+                        + getCommonOptionsString()
+                        + ");");
 
         tEnv.executeSql(
                         "INSERT INTO target "
@@ -81,6 +163,10 @@ public class OceanBaseConnectorITCase extends OceanBaseTestBase {
                                 + "       (109, 'spare tire', '24 inch spare tire', 22.2);")
                 .await();
 
+        validateSinkResults();
+    }
+
+    private void validateSinkResults() throws SQLException {
         List<String> expected =
                 Arrays.asList(
                         "101,scooter,Small 2-wheel scooter,3.1400000000",
@@ -93,20 +179,16 @@ public class OceanBaseConnectorITCase extends OceanBaseTestBase {
                         "108,jacket,water resistent black wind breaker,0.1000000000",
                         "109,spare tire,24 inch spare tire,22.2000000000");
 
-        List<String> actual = queryTable(tableName);
+        List<String> actual = queryTable();
 
         assertEqualsInAnyOrder(expected, actual);
     }
 
-    public List<String> queryTable(String tableName) throws SQLException {
+    public List<String> queryTable() throws SQLException {
         List<String> result = new ArrayList<>();
-        try (Connection connection =
-                        DriverManager.getConnection(
-                                obServer.getJdbcUrl(),
-                                obServer.getUsername(),
-                                obServer.getPassword());
+        try (Connection connection = getConnection();
                 Statement statement = connection.createStatement()) {
-            ResultSet rs = statement.executeQuery("SELECT * FROM " + tableName);
+            ResultSet rs = statement.executeQuery("SELECT * FROM " + getTestTable());
             ResultSetMetaData metaData = rs.getMetaData();
 
             while (rs.next()) {
@@ -121,5 +203,9 @@ public class OceanBaseConnectorITCase extends OceanBaseTestBase {
             }
         }
         return result;
+    }
+
+    protected Connection getConnection() throws SQLException {
+        return DriverManager.getConnection(getUrl(), getUsername(), getPassword());
     }
 }

--- a/flink-connector-oceanbase/src/test/java/com/oceanbase/connector/flink/OceanBaseConnectorITCase.java
+++ b/flink-connector-oceanbase/src/test/java/com/oceanbase/connector/flink/OceanBaseConnectorITCase.java
@@ -58,8 +58,8 @@ public class OceanBaseConnectorITCase extends OceanBaseTestBase {
     }
 
     @Override
-    protected Map<String, String> getCommonOptions() {
-        Map<String, String> options = super.getCommonOptions();
+    protected Map<String, String> getOptions() {
+        Map<String, String> options = super.getOptions();
         options.put("compatible-mode", "mysql");
         options.put("schema-name", "test");
         options.put("connection-pool-properties", "druid.initialSize=4;druid.maxActive=20;");
@@ -105,15 +105,14 @@ public class OceanBaseConnectorITCase extends OceanBaseTestBase {
                         rowData(108, "jacket", "water resistent black wind breaker", 0.1),
                         rowData(109, "spare tire", "24 inch spare tire", 22.2));
 
-        OceanBaseConnectorOptions connectorOptions =
-                new OceanBaseConnectorOptions(getCommonOptions());
-
+        OceanBaseConnectorOptions connectorOptions = new OceanBaseConnectorOptions(getOptions());
         OceanBaseConnectionProvider connectionProvider =
                 new OceanBaseConnectionPool(connectorOptions.getConnectionOptions());
-        OceanBaseTableSchema tableSchema = new OceanBaseTableSchema(physicalSchema);
         OceanBaseStatementExecutor statementExecutor =
                 new OceanBaseStatementExecutor(
-                        connectorOptions.getStatementOptions(), tableSchema, connectionProvider);
+                        connectorOptions.getStatementOptions(),
+                        new OceanBaseTableSchema(physicalSchema),
+                        connectionProvider);
         OceanBaseSink sink =
                 new OceanBaseSink(connectorOptions.getWriterOptions(), statementExecutor);
         env.fromCollection(dataSet).sinkTo(sink);
@@ -147,7 +146,7 @@ public class OceanBaseConnectorITCase extends OceanBaseTestBase {
                         + " PRIMARY KEY (`id`) NOT ENFORCED"
                         + ") with ("
                         + "  'connector'='oceanbase',"
-                        + getCommonOptionsString()
+                        + getOptionsString()
                         + ");");
 
         tEnv.executeSql(

--- a/flink-connector-oceanbase/src/test/resources/sql/init.sql
+++ b/flink-connector-oceanbase/src/test/resources/sql/init.sql
@@ -11,7 +11,8 @@
 -- specific language governing permissions and limitations
 -- under the License.
 
-use test;
+CREATE DATABASE IF NOT EXISTS test;
+USE test;
 
 CREATE TABLE products
 (

